internal-client/src/main/java/org/apache/uniffle/client/impl/grpc/ShuffleServerGrpcClient.java

```diff
@@ -212,16 +209,8 @@ private ShuffleRegisterResponse doRegisterShuffle(
         .setMaxConcurrencyPerPartitionToWrite(maxConcurrencyPerPartitionToWrite)
         .addAllPartitionRanges(toShufflePartitionRanges(partitionRanges))
         .setStageAttemptNumber(stageAttemptNumber);
-    if (StringUtils.isNotBlank(keyClassName)) {
-      reqBuilder.setKeyClass(keyClassName);
-      reqBuilder.setValueClass(valueClassName);
-      if (StringUtils.isNotBlank(comparatorClassName)) {
-        reqBuilder.setComparatorClass(comparatorClassName);
-      }
-      reqBuilder.setMergedBlockSize(mergedBlockSize);
-      if (StringUtils.isNotBlank(mergeClassLoader)) {
-        reqBuilder.setMergeClassLoader(mergeClassLoader);
-      }
+    if (mergeContext != null && StringUtils.isNotBlank(mergeContext.getKeyClass())) {
```

Comment:
Since we have check keyClass in TezRemoteShuffleManager, there is no need to check KeyClass.

Reply:
Done